ibis/backends/flink/compiler/core.py

```diff
@@ -46,15 +49,53 @@ class FlinkCompiler(Compiler):
 
 def translate(op: ops.TableNode) -> str:
     # TODO(chloeh13q): support translation of non-select exprs (e.g. literals)
-    ast = FlinkCompiler.to_ast(op)
-    return translate_language(ast.queries[0])
+    return translate_op(op)
 
 
 @functools.singledispatch
-def translate_language(language: DML | DDL) -> str:
-    raise com.OperationNotDefinedError(f'No translation rule for {type(language)}')
-
-
-@translate_language.register(Select)
-def _select(language: Select):
-    return language.compile()
+def translate_op(op: ops.TableNode) -> str:
+    raise com.OperationNotDefinedError(f'No translation rule for {type(op)}')
+
+
+@translate_op.register(ops.Literal)
+def _literal(op: ops.Literal) -> str:
+    value = op.value
+    dtype = op.output_dtype
+
+    if dtype.is_boolean():
+        # TODO(chloeh13q): Flink supports a third boolean called "UNKNOWN"
+        return 'TRUE' if value else 'FALSE'
+    elif dtype.is_string():
+        quoted = value.replace("'", "''").replace("\\", "\\\\")
+        return f"'{quoted}'"
+    elif dtype.is_date():
+        if isinstance(value, datetime.date):
+            value = value.strftime('%Y-%m-%d')
+        return repr(value)
+    elif dtype.is_numeric():
+        if math.isnan(value):
+            raise ValueError("NaN is not supported in Flink SQL")
+        elif math.isinf(value):
+            raise ValueError("Infinity is not supported in Flink SQL")
+        return repr(value)
+    elif dtype.is_timestamp():
+        # TODO(chloeh13q): support timestamp with local timezone
+        if isinstance(value, datetime.datetime):
+            return 'TIMESTAMP ' + repr(value.strftime('%Y-%m-%d %H:%M:%S'))
```

Comment:
Flink parses from format `"yyyy-MM-dd HH:mm:ss[.SSS]"`, so I think milliseconds should be included, if available.